### PR TITLE
use self-closing on vue components in template

### PR DESCRIPTION
**Pull request by 66eli77**
component self-closing tag is new in Vue2 https://vuejs.org/v2/guide/components.html#Component-Naming-Conventions

### CODE REVIEW

**Comment by christianmemije** [comment]
@66eli77 Did you lint this?

**Comment by 66eli77** [comment]
oops, I forgot to lint

**Comment by christianmemije** [comment]
Btw, I don't this will work 

> Again, this only works within string templates, as self-closing custom elements are not valid HTML and your browser’s native parser will not understand them.

**Comment by 66eli77** [comment]
templates inside .vue is string template

File: frontend_build/src/htmlhint_custom.js [hunk 1]
```diff
@@ -16,12 +16,16 @@ function clean(val) {
   Based on the existing `tag-self-close` rule
 */
 HTMLHint.addRule({
-  id: '--no-tag-self-close',
-  description: 'Self-closing tags are not valid HTML5.',
+  id: '--no-self-close-common-html5-tags',
+  description: 'Self-closing HTML5 tags are not valid.',
   init: function(parser, reporter) {
     var self = this;
+    var commonTags = parser.makeMap(`a,audio,b,body,button,canvas,caption,center,dir,div,dl,em,font,footer,
```
Comment (christianmemije) [inline frontend_build/src/htmlhint_custom.js:23]: :thinking:

Comment (christianmemije) [reply frontend_build/src/htmlhint_custom.js:23]: Gets the job done.

**Comment by christianmemije** [comment]
@66eli77 Might as well do the same for SVGs ie,
`<svg src="icon.svg"></svg>` -> `<svg src="icon.svg"/>`

**Comment by 66eli77** [comment]
@christianmemije  good point!

**Comment by christianmemije** [comment]
You missed 2:
core-base.vue: ` <error-box v-if="error"></error-box>`
task-status.vue: `<progress max="1" :value="percentage"></progress>`

**Comment by christianmemije** [comment]
@66eli77 It also looks like slots can be self-closed, even though I didn't see it get mentioned in the documentation.

**Comment by 66eli77** [comment]
@christianmemije Thanks for the catch! 
that progress is a html tag not vue component.

**Comment by christianmemije** [comment]
@66eli77 Right. For a second I thought it was the progress-bar component. 

Looks like all the cases are handled.

**Pull request merged**